rskj-core/src/test/java/co/rsk/net/sync/PeersInformationTest.java

```diff
@@ -163,4 +255,65 @@ private PeersInformation setupTopBestScenario(double topBest) {
+        trustedSnapPeersMap.put(snapPeer5.getPeerNodeID().toString(),new Node(HexUtils.strHexOrStrNumberToByteArray("0xA9E"), "snapPeer5.com", 0));
+
+        Mockito.when(blockchain.getStatus()).thenReturn(blockChainStatus);
+
+        Mockito.when(channelManager.getActivePeers()).thenReturn(Stream.of(
+                snapPeer1, snapPeer2, snapPeer3, snapPeer4
+        ).collect(Collectors.toList()));
+
+        Mockito.when(syncConfiguration.getExpirationTimePeerStatus())
+                .thenReturn(Duration.of(1, ChronoUnit.HOURS));
+
+        Mockito.when(syncConfiguration.getTopBest())
+                .thenReturn(topBest);
+
+        Mockito.when(syncConfiguration.getNodeIdToSnapshotTrustedPeerMap()).thenReturn(trustedSnapPeersMap);
+        Mockito.when(syncConfiguration.getSnapshotTrustedPeers()).thenReturn(new ArrayList<>(trustedSnapPeersMap.values()));
+
+        Mockito.when(peerScoringManager.hasGoodReputation(Mockito.eq(snapPeer1.getPeerNodeID()))).thenReturn(true);
+        Mockito.when(peerScoringManager.hasGoodReputation(Mockito.eq(snapPeer2.getPeerNodeID()))).thenReturn(true);
+        Mockito.when(peerScoringManager.hasGoodReputation(Mockito.eq(snapPeer3.getPeerNodeID()))).thenReturn(true);
+        Mockito.when(peerScoringManager.hasGoodReputation(Mockito.eq(snapPeer4.getPeerNodeID()))).thenReturn(false);
+
+        PeersInformation peersInformation = new PeersInformation(channelManager, syncConfiguration, blockchain, peerScoringManager, random);
+
+        SyncPeerStatus syncSnapPeerStatus1 = peersInformation.registerPeer(snapPeer1);
+        syncSnapPeerStatus1.setStatus(new Status(10L, "".getBytes(), null, new BlockDifficulty(BigInteger.valueOf(10L))));
+        Mockito.when(blockChainStatus.hasLowerTotalDifficultyThan(Mockito.eq(syncSnapPeerStatus1.getStatus()))).thenReturn(true);
+
+        SyncPeerStatus syncSnapPeerStatus2 = peersInformation.registerPeer(snapPeer2);
+        syncSnapPeerStatus2.setStatus(new Status(20L, "".getBytes(), null, new BlockDifficulty(BigInteger.valueOf(20L))));
+        Mockito.when(blockChainStatus.hasLowerTotalDifficultyThan(Mockito.eq(syncSnapPeerStatus2.getStatus()))).thenReturn(true);
+
+        SyncPeerStatus syncSnapPeerStatus3 = peersInformation.registerPeer(snapPeer3);
+        syncSnapPeerStatus3.setStatus(new Status(30L, "".getBytes(), null, new BlockDifficulty(BigInteger.valueOf(30L))));
+        Mockito.when(blockChainStatus.hasLowerTotalDifficultyThan(Mockito.eq(syncSnapPeerStatus3.getStatus()))).thenReturn(false);
+
+        SyncPeerStatus syncSnapPeerStatus4 = peersInformation.registerPeer(snapPeer4);
+        syncSnapPeerStatus4.setStatus(new Status(40L, "".getBytes(), null, new BlockDifficulty(BigInteger.valueOf(40L))));
+        Mockito.when(blockChainStatus.hasLowerTotalDifficultyThan(Mockito.eq(syncSnapPeerStatus4.getStatus()))).thenReturn(true);
+
+        return peersInformation;
+    }
```

Comment:
nitpick:

I am a big fan of DAMP vs DRY when it's testing code. Buuut, this method here can be easily simplified if you extract the repeated parts. Maybe you can create a map/list with the nodeIds name or id. Up to you.

Reply:
Removed repeted code by adding the [setupPeer() method](https://github.com/rsksmart/rskj/pull/2504/files#diff-f272dff0d755207a43e2091646c3c29d2db1c1a4d4553d595b58674e71b10b4fR257)